pkg/spanconfig/spanconfigsqltranslator/sqltranslator.go

```diff
@@ -366,43 +366,34 @@ func (s *SQLTranslator) generateSpanConfigurationsForTable(
 
 	records := make([]spanconfig.Record, 0)
 	if table.GetID() == keys.DescriptorTableID {
-		// We have some special handling for `system.descriptor` on account of
-		// it being the first non-empty table in every tenant's keyspace.
+		// We have named ranges preceding `system.descriptor`.
+		// Not doing anything special here would mean
+		// splitting on /Table/3 instead of /Table/0 which is benign since there's
+		// no data under /Table/{0-2}.
+		// Still, doing it this way reduces the
+		// differences between the gossip-backed subsystem and this one --
```

Comment:
nit: reword to remove the reference to "gossip-backed subsystem", or comparisons to subsystems. Just say that we have named ranges (liveness, meta) before the first table in the system tenant keyspace, so we pick the first table ID. That doesn't apply to the tenant keyspace so we can simply split at the tenant prefix itself.